### PR TITLE
feat: add options for mongo client connection

### LINKED ISSUE #857: Allow setting further connection options for MongoDB client
Hello!

Currently, the internally used MongoDB client is instantiated only with a "url" parameter, see [here](https://github.com/TNG/momo-scheduler/blob/v2.2.0/src/Connection.ts#L30). This makes it impossible to set other client options.

To use certificate based authentication for the MongoDB connection, see e.g. [these docs](https://www.mongodb.com/docs/manual/core/security-x.509/), one has to configure options like `tls`, `secureContext` (from certificate / key), `authSource` and `authMechanism`. Most of these can also be set in the [connection string options](https://www.mongodb.com/docs/manual/reference/connection-string-options/), but e.g. for certificates the connection string options only allow a file path to a certificate / key via `tlsCAFile` and `tlsCertificateKeyFile` and not passing the whole certificate itself, e.g. using a `secureContext`, [docs](https://www.mongodb.com/docs/drivers/node/current/fundamentals/connection/tls/#securecontext-example).

All in all, I think momo-scheduler should allow to pass down `MongoClientOptions` to the internal MongoDB client. I could imagine this to be done by extending the current interface `MomoConnectionOptions` [here](https://github.com/TNG/momo-scheduler/blob/v2.2.0/src/Connection.ts#L6-L15) accordingly:

```
export interface MomoConnectionOptions {
  url: string;
  options?: MongoClientOptions;
  collectionsPrefix?: string;
}
```

What do you think about it?

### PR DESCRIPTION
**Pull request by luipe**
Resolves #857

### CODE REVIEW

File: test/Connection.spec.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  it('should use provided options for the connection to the mongo client', async () => {
+    const mongoClientOptions: MongoClientOptions = {
+      tls: true,
+      secureContext: { context: 'very secure' },
+    };
+
+    await Connection.create({ url, mongoClientOptions }, 0, scheduleId, scheduleName);
+
+    expect(MongoClient).toHaveBeenCalledTimes(1);
+    expect(MongoClient).toHaveBeenCalledWith(url, mongoClientOptions);
```
[the remark this answers is not in the document]
Comment (YaniKolev) [reply test/Connection.spec.ts:22]: Yeah, they're not perfect, but I think it's correct. You are explicitly testing that we only create one mongo client and that it's the correct mongo client.

**Review by YaniKolev — APPROVED** [review]
Looks good! It should be completely backwards compatible, so I don't see the need for a major version or anything.

File: src/Connection.ts [hunk 2]
```diff
@@ -16,18 +22,18 @@ export interface MomoConnectionOptions {
 
 export class Connection {
   private constructor(
-    private readonly mongoClient: MongoClient,
+    public readonly mongoClient: MongoClient,
```
Comment (steinfurt) [inline src/Connection.ts:25]: This probably does not need to be `public`, right?

Comment (luipe) [reply src/Connection.ts:25]: Good catch, thank you! I accidentally commited this. It is a relic of my attempts to write an integration test for this

**Pull request merged**